homeassistant/components/minecraft_server/sensor.py

```diff
@@ -137,24 +136,25 @@ class MinecraftServerSensorEntity(MinecraftServerEntity, SensorEntity):
 
     def __init__(
         self,
-        server: MinecraftServer,
+        coordinator: MinecraftServerCoordinator,
         description: MinecraftServerSensorEntityDescription,
     ) -> None:
         """Initialize sensor base entity."""
-        super().__init__(server)
+        super().__init__(coordinator)
         self.entity_description = description
-        self._attr_unique_id = f"{server.unique_id}-{description.key}"
+        self._attr_unique_id = f"{coordinator.unique_id}-{description.key}"
 
     @property
-    def available(self) -> bool:
-        """Return sensor availability."""
-        return self._server.online
-
-    async def async_update(self) -> None:
-        """Update sensor state."""
-        self._attr_native_value = self.entity_description.value_fn(self._server.data)
+    def native_value(self) -> StateType:
+        """Return sensor value."""
+        return self.entity_description.value_fn(self._coordinator.data)
 
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return sensor state attributes, if available."""
+        extra_state_attributes = None
         if self.entity_description.attributes_fn:
-            self._attr_extra_state_attributes = self.entity_description.attributes_fn(
-                self._server.data
+            extra_state_attributes = self.entity_description.attributes_fn(
+                self._coordinator.data
             )
+        return extra_state_attributes
```

Comment:
```suggestion
    @callback
    def _handle_coordinator_update(self) -> None:
        """Handle updated data from the coordinator."""
        self._attr_native_value = self.entity_description.value_fn(self.coordinator.data)

        if func := self.entity_description.attributes_fn:
            self._attr_extra_state_attributes = func(self.coordinator.data)
        self.async_write_ha_state()
```

The coordinator entity supports a update callback similar to the previous code.

Reply:
Nice, thanks 👍  Should I also use the callback function in the binary sensor platform instead of the properties `available `and `is_on`?

Reply:
After testing the proposal I noticed, that all sensor entities remain "Unknown" on the first update, although the debug log from the coordinator indicates a successful update. On the second update everything is back to normal. Any idea why this is the case?

Reply:
I would leave the binary sensor as it is, as we can't use the shorthand notation for the available property there.